cpp-ch/local-engine/Storages/SubstraitSource/ReadBufferBuilder.cpp

```diff
@@ -496,16 +527,20 @@ class S3FileReadBufferBuilder : public ReadBufferBuilder
         if (!getSetting(settings, bucket_name, BackendInitializerUtil::HADOOP_S3_ASSUMED_ROLE, "", true).empty())
             is_per_bucket = true;
 
-        if (is_per_bucket && per_bucket_clients.contains(bucket_name)
-            && "true" != getSetting(settings, bucket_name, BackendInitializerUtil::HADOOP_S3_CLIENT_CACHE_IGNORE))
+        if (is_per_bucket && "true" != getSetting(settings, bucket_name, BackendInitializerUtil::HADOOP_S3_CLIENT_CACHE_IGNORE))
         {
-            return per_bucket_clients.get(bucket_name).get();
+            auto client = per_bucket_clients.get(bucket_name);
+            if (client.has_value())
```

Comment:
when the client is 'boost::none', can it call has_value() function?

Reply:
if it's none, has_value should return false?

Reply:
```
    boost::optional<size_t> test_optial = boost::none;
    boost::optional<size_t> aaa = 23;
    std::cout << test_optial.has_value() << std::endl;
    std::cout << aaa.has_value() << std::endl;
```
test code, it works.